pkg/syncer/syncer.go

```diff
@@ -130,7 +131,12 @@ func StartSyncer(ctx context.Context, cfg *SyncerConfig, numSyncerThreads int, i
 		return err
 	}
 
-	kcpInformerFactory := kcpinformers.NewSharedInformerFactoryWithOptions(kcpClient, resyncPeriod)
+	// kcpInformerFactory to watch a certain syncerTarget
+	kcpInformerFactory := kcpinformers.NewSharedInformerFactoryWithOptions(kcpClient, resyncPeriod, kcpinformers.WithTweakListOptions(
```

Comment:
One final nit - can we make sure not to use `kcpClient` for anything - right now I think the resource sync controller uses it (as opposed to going through the VW?)

Reply:
this kcpClient is scoped in synctarget workspace https://github.com/kcp-dev/kcp/blob/main/pkg/syncer/syncer.go#L89. resource sync controller needs to update condition of synctarget. But we should not use kcpClusterClient there. Updated.